service/controller/resource/s3object/update.go

```diff
@@ -82,7 +83,31 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		}
 
 		currentObject := currentS3Object[key]
-		if currentObject.Body != "" && bucketObject.Body != currentObject.Body {
+		if currentObject.Body != "" {
+			currentEncoded := base64.StdEncoding.EncodeToString([]byte(currentObject.Body))
+			currentHash := hashIgnition(currentObject.Body)
+			currentDecrypted, err := r.cloudConfig.DecryptTemplate(ctx, currentObject.Body)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			currentDecryptedHash := hashIgnition(currentDecrypted)
+			currentDecryptedEncoded := base64.StdEncoding.EncodeToString([]byte(currentDecrypted))
```

Comment:
I guess this is just for testing? We should not expose anything decrypted only with b64 encoding in our logs. That feels like a dangerous thing to do.

Reply:
Definitely! Just some debug logging.